src/core/aws-net-websocket-registry.ads

```diff
@@ -176,10 +176,18 @@ package AWS.Net.WebSocket.Registry is
       Message : String;
       Timeout : Duration := Forever;
       Error   : Error_Type := Normal_Closure);
+   --  Send a close message to the WebSocket, unregister it and release all
+   --  associated memory.
 
    function Is_Registered (Id : UID) return Boolean;
    --  Returns True if the WebSocket Id is registered and False otherwise
 
+   procedure Free (WebSocket : in out Object'Class);
+   --  Free the WebSocket when not used anymore (Either free immediatly or
+   --  register a deferred free). The difference with Close is that no message
+   --  is sent to the peer. The WebSocket is simply unregistered and memory is
+   --  released when possible.
```

Comment:
The documentation definitely helps here. But it raises the question why we need the two `Free` and `Close` subprograms. Why not simply have `Close` with an empty message if you do not want to send a message to the peer ? The goal is to reduce the size of the public API, perhaps to help users better understand what they are supposed to call (I know it wasn't clear to me when I started using websockets three years ago, and is still not 100% clear)